js/core.js

```diff
@@ -1669,52 +1669,10 @@
             //This check has been added to show "null" in all the rows if the user inputs blank string
             //We are opting json out here because we want null in the UI instead of "", so we do not call _getNullValue for json
             if (data === undefined || (data === null && this.type.name.indexOf('json') !== 0)) {
-                    return this._getNullValue(options ? options.context : undefined);
-                }
-            if (data === undefined) {
                 return this._getNullValue(options ? options.context : undefined);
             }
-            /* TODO format the raw value based on the column definition
-             * type, heuristics, annotations, etc.
-             */
-            var type = this.type.name;
-            var utils = module._formatUtils;
-            switch(type) {
-                case 'timestamptz':
-                    data = utils.printTimestamp(data, options);
-                    break;
-                case 'date':
-                    data = utils.printDate(data, options);
-                    break;
-                case 'numeric':
-                case 'float4':
-                case 'float8':
-                    data = utils.printFloat(data, options);
-                    break;
-                case 'int2':
-                case 'int4':
-                case 'int8':
-                    data = utils.printInteger(data, options);
-                    break;
-                case 'boolean':
-                    data = utils.printBoolean(data, options);
-                    break;
-                case 'markdown':
-                    // Do nothing as we will format markdown at the end of format
-                    break;
-                case 'gene_sequence':
-                    data = utils.printGeneSeq(data, options);
-                    break;
-                //Cases to support json and jsonb columns
-                case 'json':
-                case 'jsonb':
-                    data = utils.printJSON(data);
-                    break;
-                default: // includes 'text' and 'longtext' cases
-                    data = utils.printText(data, options);
-                    break;
-            }
-            return data.toString();
+
+            return _formatValueByType(this.type, data, options).toString();
```

Comment:
Why are you calling the `toString` here? If the function is expected to return string (based on jsdoc its return type is string), it should be inside that function and not here.